internal/entitlement/postgresadapter/entitlement.go

```diff
@@ -109,9 +111,18 @@ func (a *entitlementDBAdapter) CreateEntitlement(ctx context.Context, entitlemen
 	return mapEntitlementEntity(res), nil
 }
 
+func (a *entitlementDBAdapter) DeleteEntitlement(ctx context.Context, entitlementID models.NamespacedID) error {
+	_, err := a.db.Entitlement.Update().
+		Where(db_entitlement.ID(entitlementID.ID), db_entitlement.Namespace(entitlementID.Namespace)).
```

Comment:
Maybe we can consider throwing an error if the entitlement has been already deleted.